src/subclass/cell_renderer.rs

```diff
@@ -0,0 +1,658 @@
+    ) -> (i32, i32) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            if let Some(f) = (*parent_class).get_preferred_width_for_height {
+                let mut minimum_size = mem::MaybeUninit::uninit();
+                let mut natural_size = mem::MaybeUninit::uninit();
+                f(
+                    renderer.to_glib_none().0,
+                    widget.as_ref().to_glib_none().0,
+                    height,
+                    minimum_size.as_mut_ptr(),
+                    natural_size.as_mut_ptr(),
+                );
+                let minimum_size = minimum_size.assume_init();
+                let natural_size = natural_size.assume_init();
+                (minimum_size, natural_size)
+            } else {
+                (-1, -1)
+            }
+        }
+    }
+    fn parent_get_preferred_height<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+    ) -> (i32, i32) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            if let Some(f) = (*parent_class).get_preferred_height {
+                let mut minimum_size = mem::MaybeUninit::uninit();
+                let mut natural_size = mem::MaybeUninit::uninit();
+                f(
+                    renderer.to_glib_none().0,
+                    widget.as_ref().to_glib_none().0,
+                    minimum_size.as_mut_ptr(),
+                    natural_size.as_mut_ptr(),
+                );
+                let minimum_size = minimum_size.assume_init();
+                let natural_size = natural_size.assume_init();
+                (minimum_size, natural_size)
+            } else {
+                (-1, -1)
+            }
+        }
+    }
+    fn parent_get_preferred_height_for_width<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        width: i32,
+    ) -> (i32, i32) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            if let Some(f) = (*parent_class).get_preferred_height_for_width {
+                let mut minimum_size = mem::MaybeUninit::uninit();
+                let mut natural_size = mem::MaybeUninit::uninit();
+                f(
+                    renderer.to_glib_none().0,
+                    widget.as_ref().to_glib_none().0,
+                    width,
+                    minimum_size.as_mut_ptr(),
+                    natural_size.as_mut_ptr(),
+                );
+                let minimum_size = minimum_size.assume_init();
+                let natural_size = natural_size.assume_init();
+                (minimum_size, natural_size)
```

Comment:
If you want you can move the `assume_init()` calls directly into the return expression

Reply:
Not just here but everywhere :)

Reply:
done